osal/linux/osal.h

```diff
@@ -0,0 +1,59 @@
+/*
+ *  Copyright 2024 Cisco Systems, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef __OSAL_H
+#define __OSAL_H
+#include <stdarg.h>
```

Comment:
We probably have too many headers included here. There are more files included than types defined. 

I understand stdint.h signal.h, semaphore.h, select.h, pthread.h, socket.h and netinit/in.h
I think ssize_t is included from stdio.h but that remains to be checked

I think all the others could be removed or moved to the C file.